sql/gcalc_slicescan.cc

```diff
@@ -102,7 +102,7 @@ const char *gcalc_ev_name(int ev)
 
 static int gcalc_pi_str(char *str, const Gcalc_heap::Info *pi, const char *postfix)
 {
-  return sprintf(str, "%s %d %d | %s %d %d%s",
+  return snprintf(str, sizeof(str), "%s %d %d | %s %d %d%s",
```

Comment:
size of str is not known on compiling time. sizeof(str) will return the size of pointer not the size of array. I suppose it's not that you intended to do